source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTextBlockRenderer.mm

```diff
@@ -29,30 +30,41 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
 {
     std::shared_ptr<TextBlock> txtBlck = std::dynamic_pointer_cast<TextBlock>(elem);
     UILabel *lab = [[UILabel alloc] init];
-    NSString *textBlockStr = [NSString stringWithCString:txtBlck->GetText().c_str()
-                                                encoding:NSUTF8StringEncoding];
-    ContainerStyle style = [viewGroup getStyle];
+    
+    // MarkDownParser transforms text with MarkDown to a html string
+    std::shared_ptr<MarkDownParser> markDownParser = std::make_shared<MarkDownParser>(txtBlck->GetText().c_str());
+    NSString *parsedString = [NSString stringWithCString:markDownParser->TransformToHtml().c_str() encoding:NSUTF8StringEncoding];
 
-    ColorsConfig &colorConfig = (style == ContainerStyle::Emphasis)?
-        config->containerStyles.emphasisPalette.foregroundColors:
-        config->containerStyles.defaultPalette.foregroundColors;
-
-    NSMutableAttributedString *content =
-    [[NSMutableAttributedString alloc] initWithString:textBlockStr
-                                           attributes:@{NSForegroundColorAttributeName:[ACRTextBlockRenderer getTextBlockColor:txtBlck->GetTextColor() colorsConfig:colorConfig subtleOption:txtBlck->GetIsSubtle()],
-                                                            NSStrokeWidthAttributeName:[ACRTextBlockRenderer getTextBlockTextWeight:txtBlck->GetTextWeight() withHostConfig:config]}];
-    NSMutableParagraphStyle *para = [[NSMutableParagraphStyle alloc] init];
-    para.lineBreakMode = txtBlck->GetWrap() ? NSLineBreakByWordWrapping:NSLineBreakByTruncatingTail;
-    para.alignment = [self getTextBlockAlignment:txtBlck withHostConfig:config];
-    [content addAttributes:@{NSParagraphStyleAttributeName:para} range:NSMakeRange(0,1)];
-    lab.attributedText = content;
+    // Font and text size are applied as CSS style by appending it to the html string -- font is hard coded for now
+    UIFontDescriptor *dec = lab.font.fontDescriptor;
+    UIFont *labelFont = [UIFont fontWithDescriptor:dec size:[ACRTextBlockRenderer getTextBlockTextSize:txtBlck->GetTextSize() withHostConfig:config]];
+    parsedString = [parsedString stringByAppendingString:[NSString stringWithFormat:@"<style>body{font-family: '%@'; font-size:%fpx;}</style>",
```

Comment:
>px [](start = 129, length = 2)

Should this be `pt` to match `label.pointSize`? Points and pixels are different units.

Reply:
Yes, you are correct. Thanks for finding this.
Since I'm using CSS to configure font, I don't need to get UIFont anymore, so I will use the font size directly using GetTextSize() which returns the size in pix. Thanks!

---
In reply to: [159497690](https://github.com/Microsoft/AdaptiveCards/pull/1054#discussion_r159497690) [](ancestors = 159497690)